pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -188,6 +188,8 @@ public void stop() {
       }
     }
     LOGGER.info("Wait completed: Number of completing segments = {}", _numCompletingSegments.get());
+    INSTANCE = null;
```

Comment:
Can you add TODO to fix the misuse of singleton class? This should not be a singleton class. Same for other classes.

Reply:
TODO added.